src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -752,13 +752,25 @@ private function registerValidationConfiguration(array $config, ContainerBuilder
 
         $container->setParameter('validator.translation_domain', $config['translation_domain']);
 
-        list($xmlMappings, $yamlMappings) = $this->getValidatorMappingFiles($container);
-        if (count($xmlMappings) > 0) {
-            $validatorBuilder->addMethodCall('addXmlMappings', array($xmlMappings));
+        $files = array('xml' => array(), 'yml' => array());
+        foreach ($this->getValidatorMappingFiles($config, $container) as $file) {
+            if (!is_file($file)) {
+                throw new \RuntimeException(sprintf('Could not load file "%s".', $file));
+            }
+            $container->addResource(new FileResource($file));
```

Comment:
please keep resources inside the method loading the files, because the real need is to check for the existence of the resource, not for the content of the resource. This means that user-defined files don't need to be considered as a resource (and framework-defined resources should be handled differently, which I will do in a separate PR, and so your change would make it more complex)